HighAvailabilityModule.Storage.Client/SQLStorageMembershipClient.cs

```diff
@@ -0,0 +1,392 @@
+﻿namespace HighAvailabilityModule.Storage.Client
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Data;
+    using System.Data.SqlClient;
+
+    using HighAvailabilityModule.Interface;
+    using System.Diagnostics.Tracing;
+    using System.Runtime.CompilerServices;
+    using System.Collections;
+    using System.Runtime.InteropServices;
+
+    public class SQLStorageMembershipClient : IMembershipStorageClient
+    {
+        public string ConStr { get; set; }
+
+        public TimeSpan OperationTimeout { get; set; }
+
+        private const string GetDataEntrySpName = "dbo.GetDataEntry";
+        private const string SetDataEntrySpName = "dbo.SetDataEntry";
+        private const string DeleteDataEntrySpName = "dbo.DeleteDataEntry";
+        private const string EnumerateDataEntrySpName = "dbo.EnumerateDataEntry";
+
+        public SQLStorageMembershipClient(string conStr, TimeSpan operationTimeout)
+        {
+            this.OperationTimeout = operationTimeout;
+            if (conStr.IndexOf("Connect Timeout") == -1)
+            {
+                this.ConStr = conStr + ";Connect Timeout=" + Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds)).ToString();
+            }
+            else
+            {
+                this.ConStr = conStr.Substring(0, conStr.IndexOf("Connect Timeout")) + "Connect Timeout=" + Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds)).ToString();
+            }
+        }
+
+        public async Task <(string, string)> GetDataEntryAsync(string path, string key)
```

Comment:
Set a more descriptive name for the returned tuple. like `(string value, string type)`